### PR TITLE
Node.js Package improve

### PR DESCRIPTION
**Pull request by denlb**
Publish only if the version has changed

ありがとうございます @okuryu
https://github.com/actions/setup-node/issues/53#issuecomment-534558301

### CODE REVIEW

**Comment by okuryu** [comment]
Yes, I think it works fine, but I'm not sure this change is the best as a workflow of the starter-workflows.

- To publish with this change you have to use GitHub Release feature on GitHub UI
- Customers may want to trigger an event other than `releases`
- Customers may want to create git tags first via `npm version` command

There is also a useful action called [`merge-release`](https://github.com/mikeal/merge-release) created by @mikeal which automatically updates the version according to the semantic version based on your commit log, and performs git tagging and publishing packages.

I think customer requirements vary, so proposing a common workflow is difficult.

**Comment by denlb** [comment]
@okuryu maybe you're right about releases, so i'll revert that but `$LATEST" != "$CURRENT` is a must-have otherwise you'll get your action failing

File: ci/npm-publish.yml [hunk 1]
```diff
@@ -28,7 +28,15 @@ jobs:
         with:
           node-version: 12
           registry-url: https://registry.npmjs.org/
-      - run: npm publish
+          scope: '@your-github-username'
```
Comment (okuryu) [inline ci/npm-publish.yml]: I think the scope setting is not required for `registry.npmjs.org`.

Comment (denlb) [reply ci/npm-publish.yml]: That's true, just my preference. Reverted

**Comment by andymckay** [comment]
Thank you @denlb for this pull request 🙇 

We've changed the action to only trigger on `release` when its event `types: [created]`. So hopefully this reduces occurrence of this situation. Based on that I think we can skip doing this version number check. If this doesn't fix the situation, please let us know.

**Pull request closed without merging**